exercises/practice/sum-of-multiples/.approaches/filter-for-multiples/content.md

```diff
@@ -0,0 +1,253 @@
+To use the _cursor_ analogy: after going through all of `even_number`'s 'elements' &ndash; actually elements of `numbers` &ndash; the cursor has moved all the way to the right.
+It cannot move backwards, so if you wish to iterate over all even numbers again then you need a new cursor.
+We say that the `even_numbers` iterator is _exhausted_. When `sum` asks for its elements again, `even_numbers` comes up empty and so `sum` returns `0`.
+
+Had the highlighted solution not used `filter`, it might have looked like this:
+
+```python
+def sum_of_multiples(limit, factors):
+    def is_multiple(n):
+        return any(n % f == 0 for f in factors if f != 0)
+
+    multiples = [candidate for candidate in range(limit) if is_multiple(candidate)]
+    return sum(multiples)
+```
+
+This variant stores all the multiples in a `list` before summing them.
+Such a list can become very big.
+For example, if `limit = 1_000_000_000` and `factors = [1]` then `multiples` will take up 8 gigabytes of memory!
+It is to avoid unnecessarily creating such large intermediate data structures that iterators are often used.
+
+
+### A function expression: `lambda`
+
+Typically, when using higher-order functions like `filter` and `map`, the function to pass as an argument does not yet exist and needs to be defined first.
+
+The standard way of defining functions is through the `def` statement:
+
+```python
+def name(parameters):
+    statements
+```
+
+Downsides of this construct include
+
+- the syntax can be a bit bulky
+- it requires coming up with a fresh name
+
+These qualities can be quite bothersome when you just need a simple function of no particular significance for single use only.
+In situations like this you might like to use a **lambda expression** instead.
+
+A lambda expression is a specific kind of expression that evaluates to a function.
+It looks like this:
+
+```python
+lambda parameters: expression  # general form
+lambda a, b, x: a * x + b      # specific example
+```
+
+This latter lambda expression evaluates to a function that takes three arguments (`a`, `b`, `x`) and returns the value `a * x + b`.
+Except for not having a name, it is equivalent to the function defined by
+
+```python
+def some_name(a, b, x):
+    return a * x + b
+```
+
+A lambda expression need not necessarily be passed as an argument.
+It can also be applied to arguments immediately, or assigned to a variable:
+
+```python
+lambda a, b, x: a * x + b
+# ⟹ <function <lambda> at 0x000001F36A274CC0>
+
+(lambda a, b, x: a * x + b)(2, 3, 5)
+# ⟹ 13
+
+some_function = lambda a, b, x: a * x + b
+some_function(2, 3, 5)
+# ⟹ 13
+
+list(filter(
+    lambda s: len(s) <= 3, 
+    ["aaaa", "b", "ccccc", "dd", "eee"]
+))
+# ⟹ ['b', 'dd', 'eee']
+```
+
+~~~~exercism/note
+Immediately applying a lambda expression is possible, but generally pointless:
+
+```python
+# Instead of
+(lambda a, b, x: a * x + b)(2, 3, y)
+# you might as well write
+2 * y + 3
+```
+~~~~
```

Comment:
```suggestion
```
```

Reply:
That `~~~~` is matched with the `~~~~exercism/note` 9 lines up.

Reply:
Huh.  Ok.  It was reading really weird to me.  Feel free to ignore.

Reply:
Was it the code or the rendering that read weird?

Reply:
It was the rendering, which is sorta gross on GH:

![image](https://user-images.githubusercontent.com/5923094/230168949-401adbde-5df1-4f7b-a19a-5f7e7b93e30b.png)

<br>

But it was more that I missed the first `note` block, as I wasn't expecting it.  😄   Now that I see there is a note and a `caution`, it makes sense.